lib/Tunnel.js

```diff
@@ -59,12 +69,21 @@ Tunnel.prototype._init = function(cb) {
                 max_conn: max_conn
             });
         })
-        .catch(function(err){
-            // TODO (shtylman) don't print to stdout?
-            console.log('tunnel server offline: ' + err.message + ', retry 1s');
-            return setTimeout(get_url, 1000);
+        .catch(function(attemptErr){
+            if (op.retry(attemptErr)) {
+                // TODO (shtylman) don't print to stderr?
+                console.error('failed to connect to the tunnel server, retrying', attemptErr);
+                return;
+            }
+
+            const actualErr = op.mainError();
+            const err = new Error('failed to connect to the tunnel server');
+            err.actualErr = actualErr;
+            // TODO (shtylman) don't print to stderr?
```

Comment:
Another one here, maybe just make it configurable so people can decide for themselves?

Reply:
Let's create another PR for this!